src/init.c

```diff
@@ -785,7 +785,7 @@ void jl_get_builtin_hooks(void)
     jl_errorexception_type = (jl_datatype_t*)core("ErrorException");
     jl_stackovf_exception  = jl_new_struct_uninit((jl_datatype_t*)core("StackOverflowError"));
     jl_diverror_exception  = jl_new_struct_uninit((jl_datatype_t*)core("DivideError"));
-    jl_overflow_exception  = jl_new_struct_uninit((jl_datatype_t*)core("OverflowError"));
+    jl_c_overflow_exception = jl_new_struct_uninit((jl_datatype_t*)core("COverflowError"));
```

Comment:
This seems sort of odd to have two overflow exception types. It would be nicer to use something like:
```c
jl_c_overflow_exception = core("overflow_error_from_c")
```
```julia
const overflow_error_from_c = OverflowError(ErrorException(""))
```

But there's only one active use of this in src (in jl_compute_field_offsets), so it's perhaps better just to delete this entirely and use `jl_errorf` there.